src/setup_payload/QRCodeSetupPayloadGenerator.cpp

```diff
@@ -202,16 +205,33 @@ CHIP_ERROR QRCodeSetupPayloadGenerator::payloadBase38Representation(std::string
                                                                     uint32_t tlvDataStartSize)
 {
     size_t tlvDataLengthInBytes = 0;
+    std::vector<uint8_t> bits(kTotalPayloadDataSizeInBytes);
+    MutableByteSpan bitsSpan;
+    std::vector<char> buffer;
+    MutableCharSpan bufferSpan;
 
     VerifyOrReturnError(mPayload.isValidQRCodePayload(), CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorOnFailure(generateTLVFromOptionalData(mPayload, tlvDataStart, tlvDataStartSize, tlvDataLengthInBytes));
 
-    return payloadBase38RepresentationWithTLV(mPayload, base38Representation, kTotalPayloadDataSizeInBytes + tlvDataLengthInBytes,
-                                              tlvDataStart, tlvDataLengthInBytes);
+    bits.reserve(kTotalPayloadDataSizeInBytes + tlvDataLengthInBytes);
+    bitsSpan = MutableByteSpan(bits.data(), bits.capacity());
+    buffer.reserve(base38EncodedLength(bits.capacity()) + strlen(kQRCodePrefix));
+    bufferSpan = MutableCharSpan(buffer.data(), buffer.capacity());
```

Comment:
Again, just construct `bufferSpan` here:
```suggestion
    MutableCharSpan bufferSpan(buffer.data(), buffer.capacity());
```